File: Two_points/1.two-sum.go
package leetcode

// @leet start
func twoSum(nums []int, target int) []int {
	m := make(map[int]int)
	for i, v := range nums {
		diff := target - v
		if idx, ok := m[diff]; ok {
			return []int{idx, i}
		}
		m[v] = i
	}
    
	return nil
}

// @leet end

File: Two_points/1.two-sum.ts
// @leet start
function twoSum(nums: number[], target: number): number[] {
  const m = new Map<number, number>()
  for( const [idx, v] of nums.entries()){
    const diff = target - v
    if( m.has(diff)){
      return [idx, m.get(diff)!]
    }
    m.set(v, idx)
  }
  return []
};
// @leet end
